internal/state/rollback.go

```diff
@@ -65,10 +65,11 @@ func Rollback(bs BlockStore, ss Store, removeBlock bool) (int64, []byte, error)
 		return -1, nil, err
 	}
 
+	nextHeight := rollbackHeight + 1
 	valChangeHeight := invalidState.LastHeightValidatorsChanged
 	// this can only happen if the validator set changed since the last block
-	if valChangeHeight > rollbackHeight {
-		valChangeHeight = rollbackHeight + 1
+	if valChangeHeight > nextHeight+1 {
+		valChangeHeight = nextHeight + 1
```

Comment:
If I run the unit tests changing this line with this:
```go
valChangeHeight = nextHeight + 2
```
which is the correct one?

Reply:
`valChangeHeight = nextHeight + 1` is right, if it changed to `valChangeHeight = nextHeight + 2`, will cause error in some edge case